### ISSUE TITLE
Unable to log param

### ISSUE THREAD

**Issue by LeanderK** [mlflow/mlflow#877]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **MLflow installed from (source or binary)**: pip
- **MLflow version (run ``mlflow --version``)**: mlflow, version 0.8.2
- **Python version**: Python 3.6.8 :: Anaconda, Inc.
- **npm version (if running the dev UI): - 
- **Exact command to reproduce**: (not so easy)

### Describe the problem
I am unable to log a parameter. I want to log a string-representation of my pytorch-model. It's a larger string. 
The code is:

```python
    print(model.extra_repr())
    start_time = time()
    import pdb; pdb.set_trace()
    with mlflow.start_run():
        mlflow.log_param("cuda", use_cuda)
        mlflow.log_param("architecture", model.extra_repr())
        mlflow.log_param("epochs", epochs)
        mlflow.log_param("in-class", in_class)
        mlflow.log_param("all-classes", all_classes)
        mlflow.log_param("contamination", contamination)
        mlflow.log_param("roc", "true")
        mlflow.log_param("batch_size", batch_size)
        mlflow.log_param("optimizer", optimizer.__class__.__name__)
        mlflow.log_param("lr", opt_lr)
        # for key, value in vars(args).items():
        #    if value is not '':
        #        mlflow.log_param(key, value)
        train_with_roc(model, device, optimizer, epochs, writer, model_dir, train_loader, val_in_loader, val_out_loader,
                       test_in_loader, test_out_loader, max_iter_val=10, max_iter_test=100, do_plot_input_samples=True, use_ddi=False)
        print("finished training!")
    mlflow.end_run()
```
the type of `mode.extra_rep()` is a string.

traceback:
```python
Traceback (most recent call last):
  File "models/naive-mnist-v2.py", line 185, in <module>
    print("finished training!")
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/tracking/fluent.py", line 65, in __exit__
    end_run(status)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/tracking/fluent.py", line 157, in end_run
    MlflowClient().set_terminated(_active_run_stack[-1].info.run_uuid, status)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/tracking/client.py", line 213, in set_terminated
    end_time=end_time)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 293, in update_run_info
    run_info = self.get_run(run_uuid).info
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 355, in get_run
    params = self.get_all_params(run_uuid)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 474, in get_all_params
    params.append(self._get_param_from_file(parent_path, param_file))
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 452, in _get_param_from_file
    % param_name)
Exception: Unexpected data for param 'architecture'. Param recorded more than once
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/tracking/fluent.py", line 157, in end_run
    MlflowClient().set_terminated(_active_run_stack[-1].info.run_uuid, status)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/tracking/client.py", line 213, in set_terminated
    end_time=end_time)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 293, in update_run_info
    run_info = self.get_run(run_uuid).info
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 355, in get_run
    params = self.get_all_params(run_uuid)
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 474, in get_all_params
    params.append(self._get_param_from_file(parent_path, param_file))
  File "(just a normal path)/lib/python3.6/site-packages/mlflow/store/file_store.py", line 452, in _get_param_from_file
    % param_name)
Exception: Unexpected data for param 'architecture'. Param recorded more than once
```

**Comment by dstine** [comment]
Just a guess: the `model.extra_repr()` string has new lines in it.

[The code](https://github.com/mlflow/mlflow/blob/v0.8.2/mlflow/store/file_store.py#L450-L452) in for `_get_param_from_file` in `file_store.py` implies that each line of the parameter has been interpreted as a separate "value" for the parameter. The error check is probably for the reason it says (logging a parameter more than once) but in this case is misleading.

**Comment by LeanderK** [comment]
hmm, yes, it has newlines in it. Otherwise the string would be enourmous. Is there any way to disable that behaviour? I find it highly unintuitve, also...does it even make sense since you can't pass multiple values to a parameter?

**Comment by dstine** [comment]
Preface: I'm just a new user, so you should see what the project team says.

Having said that, my sense is that MLflow has a design principle that parameters are "small". They get displayed in the UI, they can be queried in the UI, etc. A consequence of that might be that newlines are not supported. (Though if that is the case, it could be made more explicit.)

You might be able to log the model representation as an artifact, though I don't know if that would satisfy your use case.

**Comment by LeanderK** [comment]
ok, thought you were part of the team. Well, it's not enormous from a pure memory-usage point-of-view. But displaying it in one line would be not a solution. A text-file would be possible, but I would like to see them in the UI. I don't know whether that's possible (I can't start the UI due to #873 )

**Comment by smurching** [comment]
@LeanderK good question, the right feature/approach for this use case is probably to log the model architecture as an artifact - once you click into the UI for an individual run, you can see the text of an artifact, as long as its file extension is one of `['txt', 'log', 'py', 'js', 'yaml', 'yml', 'json', 'csv', 'tsv', md', 'rst', 'MLmodel', 'MLproject']`.

See this screenshot for an example of the artifact UI:
![image](https://user-images.githubusercontent.com/2358483/52465501-a8649000-2b33-11e9-8221-43b47d86e3cf.png)


To address the specific issue - multi-line params aren't currently supported in the FileStore (this limitation is also discussed in #614) but will be supported when logging to a SQL-backed store, which will be released with MLFlow 0.9.0. We could considering adding support for multi-line parameters as we did for tags in https://github.com/mlflow/mlflow/pull/396, but I don't think the current UI would handle them well (it'd truncate the param values after the first line) and I'm not sure what the use case is for logging multi-line param values.

**Comment by LeanderK** [comment]
ok, thank you for your help!
